File: backend/src/main/java/com/example/demo/dto/LeccionDTO.java
package com.example.demo.dto;

import java.util.List;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LeccionDTO {
    private Long id;
    private String titulo;
    private String descripcion;

    private Long moduloId;
    private String moduloTitulo;

    private List<MaterialDTO> materiales;
    private List<EvaluacionDTO> evaluaciones;
}

File: frontend/src/app/models/leccion.dto.ts
// src/app/models/leccion.dto.ts

import { EvaluacionDTO } from "./evaluacion.dto";
import { MaterialDTO } from "./material.dto";

export interface LeccionDTO {
  id?: number;
  titulo: string;
  descripcion?: string;
  moduloId?: number;
  moduloTitulo?: string;

  materiales?: MaterialDTO[];

  evaluaciones?: EvaluacionDTO[];  // ✔ AHORA sí
}